python/src/trezorlib/debuglink.py

```diff
@@ -2135,4 +2146,68 @@ def _letter_coords_and_amount(self, letter: str) -> t.Tuple[Coords, int]:
             i for i, letters in enumerate(BUTTON_LETTERS_BIP39) if letter in letters
         )
         click_amount = BUTTON_LETTERS_BIP39[idx].index(letter) + 1
-        return self.buttons.mnemonic_from_index(idx), click_amount
+        return self.debuglink.screen_buttons.mnemonic_from_index(idx), click_amount
+
+    def navigate_to_menu_item(self, idx: int) -> None:
+        """Navigate to the nth item in the vertical menu. Starts from 0."""
+        layout = self.debuglink.read_layout()
+        if self.debuglink.layout_type is LayoutType.Delizia:
+            # fido multi-screen menu with 2 items per screen
+            if "PagedVerticalMenu" in layout.all_components():
+                items_per_screen = 2
+                # get to the correct screen
+                for _ in range(idx // items_per_screen):
+                    self.debuglink.swipe_up()
+                    assert (
+                        "PagedVerticalMenu"
+                        in self.debuglink.read_layout().all_components()
+                    )
+                # click the correct item
+                new_idx = 0 if idx % items_per_screen == 0 else 2
+                self.debuglink.click(
+                    self.debuglink.screen_buttons.vertical_menu_items()[new_idx]
+                )
+            elif "ScrolledVerticalMenu" in layout.all_components():
+                _prev, next = self.debuglink.screen_buttons.vertical_menu_prev_next()
+                menu = layout.find_unique_value_by_key(
+                    key="menu_items", default=None, only_type=dict
+                )
+                # multi-screen variant
+                if menu["has_next"]:
+                    items_per_screen = 2
+                    # get to the correct screen
+                    for _ in range(idx // items_per_screen):
+                        assert menu["has_next"]
+                        self.debuglink.click(next)
+                    # click the correct item
+                    assert (
+                        "ScrolledVerticalMenu"
+                        in self.debuglink.read_layout().all_components()
+                    )
+                    new_idx = idx % items_per_screen
+                    self.debuglink.click(
+                        self.debuglink.screen_buttons.vertical_menu_items()[new_idx]
+                    )
+                # single-screen variant
+                else:
+                    assert (
+                        len(self.debuglink.screen_buttons.vertical_menu_items()) > idx
+                    )
+                    self.debuglink.click(
+                        self.debuglink.screen_buttons.vertical_menu_items()[idx]
+                    )
+            # single-screen static menu
+            # FIXME: remove this when the ScrollableVerticalMenu is implemented everywhere
+            else:
+                assert len(self.debuglink.screen_buttons.vertical_menu_items()) > idx
```

Comment:
nit: `self.debuglink.screen_buttons.vertical_menu_items()` is used multiple times  in this method - consider calling it once in the beginning.